### PR TITLE
UIDATIMP-1240: Refactor profiles to use Router

### PR DESCRIPTION
**Pull request by mariia-aloshyna**
## Purpose

To refactor profiles to use Router instead of `layout` query param to avoid bugs when refreshing the page

## Approach

- Rewrite the custom `SearchAndSort` component to use Rotes to render Create/Edit/Duplicate screens
- Create `CreateProfile`, `EditProfile`, `ViewProfile` containers for each profile type

## Refs

https://issues.folio.org/browse/UIDATIMP-1240
https://issues.folio.org/browse/UIDATIMP-1279
https://issues.folio.org/browse/UIDATIMP-1280
https://issues.folio.org/browse/UIDATIMP-1281
https://issues.folio.org/browse/UIDATIMP-1282

## Screenshots

**Before:** the problem appears after the page refreshing
chrome_ytPZgvmzRx

**After:** Use Routes
chrome_qq5qdOGkeW

### CODE REVIEW

**Comment by OleksandrHladchenko1** [comment]
@folio-org/fe-tl-reviewers please, take a look

File: src/settings/MappingProfiles/EditMappingProfile/EditMappingProfile.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useIntl } from 'react-intl';
+import {
+  get,
+  omit
```
Comment (OleksandrHladchenko1) [inline src/settings/MappingProfiles/EditMappingProfile/EditMappingProfile.js:6]:
```suggestion
  omit,
```

File: src/settings/MappingProfiles/EditMappingProfile/EditMappingProfile.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+import { MappingProfilesForm } from '../MappingProfilesForm';
+
+import { LAYER_TYPES } from '../../../utils';
+
+const EditMappingProfileComponent = ({
+  resources: { mappingProfile },
+  fullWidthContainer,
+  layerType,
+  ...routeProps
```
Comment (OleksandrHladchenko1) [inline src/settings/MappingProfiles/EditMappingProfile/EditMappingProfile.js:21]:
```suggestion
  ...routeProps,
```

File: src/settings/MatchProfiles/CreateMatchProfile/CreateMatchProfile.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import { useIntl } from 'react-intl';
+
+import { Layer } from '@folio/stripes/components';
+
+import { MatchProfilesForm } from '../MatchProfilesForm';
+import { getSectionInitialValues } from '../MatchProfiles';
+
+export const CreateMatchProfile = ({
+  fullWidthContainer,
+  ...routeProps
```
Comment (OleksandrHladchenko1) [inline src/settings/MatchProfiles/CreateMatchProfile/CreateMatchProfile.js:12]:
```suggestion
  ...routeProps,
```

File: src/settings/MatchProfiles/EditMatchProfile/EditMatchProfile.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useIntl } from 'react-intl';
+import {
+  get,
+  omit
```
Comment (OleksandrHladchenko1) [inline src/settings/MatchProfiles/EditMatchProfile/EditMatchProfile.js:6]:
```suggestion
  omit,
```

File: src/settings/MatchProfiles/EditMatchProfile/EditMatchProfile.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+import { MatchProfilesForm } from '../MatchProfilesForm';
+
+import { LAYER_TYPES } from '../../../utils';
+
+const EditMatchProfileComponent = ({
+  resources: { matchProfile },
+  fullWidthContainer,
+  layerType,
+  ...routeProps
```
Comment (OleksandrHladchenko1) [inline src/settings/MatchProfiles/EditMatchProfile/EditMatchProfile.js:21]:
```suggestion
  ...routeProps,
```

File: src/settings/ActionProfiles/CreateActionProfile/CreateActionProfile.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+import PropTypes from 'prop-types';
+import { useIntl } from 'react-intl';
+
+import { Layer } from '@folio/stripes/components';
+
+import { ActionProfilesForm } from '../ActionProfilesForm';
+
+export const CreateActionProfile = ({
+  fullWidthContainer,
+  ...routeProps
```
[the remark this answers is not in the document]
Comment (mariia-aloshyna) [reply src/settings/ActionProfiles/CreateActionProfile/CreateActionProfile.js:11]: It's forbidden to use comma after the rest
image

**Comment by mariia-aloshyna** [comment]
@folio-org/fe-tl-reviewers take a look, please

**Review by BogdanDenis — APPROVED** [review]

**Review by OleksandrHladchenko1 — APPROVED** [review]

**Pull request merged**